Sources/Authentication/Header/HeaderValueAuthenticatable.swift

```diff
@@ -0,0 +1,40 @@
+
+/// Authenticatable by a value found
+/// in one of the HTTP headers.
+public protocol HeaderValueAuthenticatable: Authenticatable {
+    /// The type of authorization being used.
+    /// Most commonly this is BearerAuthorization
+    /// but ValueAuthorization can be used if you are
+    /// authenticating with a value not found in the
+    /// Authorization: Bearer header.
+    associatedtype AuthorizationType: HeaderValueAuthorization
```

Comment:
I think `Token` or `Value` could be better here. It's not really a `Type`.

Reply:
Dig it.